### ISSUE TITLE
[kube-prometheus / kube-state-metrics] Unable to scrape metrics with Istio injection enabled

### ISSUE THREAD

**Issue by leotomas837** [bitnami/charts#16086]
### Name and Version

bitnami/kube-prometheus 8.5.0, bitnami/kube-state-metrics 3.3.5

### What is the problem this feature will solve?

Unable to scrape metrics with Istio injection enabled.

When having istio injection enabled at the namespace level (namespace where bitnami/kube-prometheus with bitnami/kube-state-metrics and bitnami/node-exporter, and bitnami/thanos are all installed), the pods from the following resources cannot be scraped:

- deploy/kube-prometheus-system-operator
- deploy/kube-prometheus-system-kube-state-metrics
- sts/alertmanager-kube-prometheus-system-alertmanager

This is due to the fact that the scraped endpoints must enforce a specific http scheme and tls_config, see [the istio doc here](https://istio.io/latest/docs/ops/integrations/prometheus/#tls-settings).

### What is the feature you are proposing to solve the problem?

Doing the same as for bitnami/thanos chart, i.e.

[kube-prometheus]
Adding the following fields:
- `operator.serviceMonitor.extraParameters` --> would control the` http` endpoint of the `ServiceMonitor` of the `operator`
- `alertmanager.serviceMonitor.extraParameters` --> would control the `http` endpoint of the `ServiceMonitor` of the `alertmanager`

[kube-state-metrics]
Adding the following field: `serviceMonitor.extraParameters` -->  would control both the `http` and `metrics` endpoints of the (only) `ServiceMonitor`

No need to enable this field for other components to fix this problem. For example: the `coredns` pods are in `kube-system` namespace therefore cannot have the istio sidecar injected and the issue will never appear, etc.

With these changes, we would be able to do the following for the `kube-state-metrics` pod for example, and have its metrics scraped (following the istio doc link I shared in the previous section):

```yaml
# values.yaml
serviceMonitor:
  enabled: true
  extraParameters:
    scheme: "https"
    tlsConfig:
      caFile: /etc/prom-certs/root-cert.pem
      certFile: /etc/prom-certs/cert-chain.pem
      keyFile: /etc/prom-certs/key.pem
      insecureSkipVerify: true
``` 

### What alternatives have you considered?

Patching the serviceMonitors with a controller as [patch-operator](https://github.com/redhat-cop/patch-operator) to keep GitOps pattern at best and avoid manual changes. Not really convenient at all.

**Comment by leotomas837** [comment]
Came someone review the 2 pull requests ?

**Comment by carrodher** [comment]
Thanks for creating this issue and the associated PR. The team will review it and provide feedback. Once merged the PR, this issue will be automatically closed.

**Comment by leotomas837** [comment]
@carrodher 
Could you re-open the issue ? PR #16087 has not been merged yet, only #16088 is merged. Could you check the remaining PR?

**Comment by leotomas837** [comment]
Could this be re-open ? Still an issue: updating the kube-state-metrics dependency in the kube-prometheus chart. See PR #16146
